test/jdk/javax/accessibility/4170173/AccessibleJTextAfterIndexTest.java

```diff
@@ -0,0 +1,122 @@
+            throw new RuntimeException(
+                "JTextField - " + "getAfterIndex() WORD parameter"
+                    + "expected: Test2, actual: " + actualAccessibleText);
+        }
+        if (!checkAccessibleText(jTextField, AccessibleText.SENTENCE, 5,
+            "Test4 Test5. ")) {
+            throw new RuntimeException("JTextField - "
+                + "getAfterIndex() SENTENCE parameter"
+                + "expected: Test4 Test5. , actual: " + actualAccessibleText);
+        }
+        if (!checkAccessibleText(jTextArea, AccessibleText.CHARACTER, 5, "T")) {
+            throw new RuntimeException(
+                "JTextArea - " + "getAfterIndex() CHARACTER parameter"
+                    + "expected: T , actual: " + actualAccessibleText);
+        }
+        if (!checkAccessibleText(jTextArea, AccessibleText.WORD, 5, "Test2")) {
+            throw new RuntimeException(
+                "JTextArea - " + "getAfterIndex() WORD parameter"
+                    + "expected: Test2 , actual: " + actualAccessibleText);
+        }
+        if (!checkAccessibleText(jTextArea, AccessibleText.SENTENCE, 5,
+            "Test4 Test5\n")) {
+            throw new RuntimeException("JTextArea - "
+                + "getAfterIndex() SENTENCE parameter"
+                + "expected: Test4 Test5\n, actual: " + actualAccessibleText);
+        }
+        if (!checkAccessibleText(jEditorPane, AccessibleText.CHARACTER, 5,
+            "T")) {
+            throw new RuntimeException(
+                "JTextArea - " + "getAfterIndex() CHARACTER parameter"
+                    + "expected: T, actual: " + actualAccessibleText);
+        }
+        if (!checkAccessibleText(jEditorPane, AccessibleText.WORD, 5,
+            "Test2")) {
+            throw new RuntimeException(
+                "JEditorPane - " + "getAfterIndex() WORD parameter"
+                    + "expected: Test2, actual: " + actualAccessibleText);
+        }
+        if (!checkAccessibleText(jEditorPane, AccessibleText.SENTENCE, 5,
+            "Test4 Test5\n")) {
+            throw new RuntimeException("JEditorPane - "
+                + "getAfterIndex() Sentence parameter"
+                + "expected: Test4 Test5\n, actual: " + actualAccessibleText);
+        }
+    }
+
+    private static void createGUI() {
+        jTextField = new JTextField("Test1 Test2 Test3. Test4 Test5. Test6");
+        jTextArea = new JTextArea("Test1 Test2 Test3.\nTest4 Test5");
+        jEditorPane =
+            new JEditorPane("text/plain", "Test1 Test2 Test3.\nTest4 Test5");
+    }
+
+    private static boolean checkAccessibleText(JTextComponent jTextComponent,
+        int character, int characterPosition, String expectedString)
+            throws Exception {
+        SwingUtilities.invokeAndWait(() -> actualAccessibleText =
+            jTextComponent.getAccessibleContext().getAccessibleText()
+            .getAfterIndex(character, characterPosition));
+
```

Comment:
It feels that the code above is too smart=) Can we simplify/reformat it a little bit. I do not think that you need to jump from the main thread to the EDT and back for each testcase, just wrap the "doTest" by the invokeAndWait

Reply:
This is a really good suggestion, thank you. I have simplified the test.